src/python/regtech_regex/regex_config.py

```diff
@@ -0,0 +1,37 @@
+import os
+from re import Pattern, compile
+from dataclasses import dataclass
+from typing import List
+
+import yaml
+
+
+@dataclass
+class RegexConfig:
+    description: str
+    error_text: str
+    regex: Pattern
+    examples: List[str] | None = None
+    link: str | None = None
+    references: List[str] | None = None
+
+
+class ConfigFactory:
```

Comment:
I'm wondering if we need to use the factory pattern for this; if we do, I believe this should become either a static class or use a singleton. The `get_regex_configs` shouldn't need to be explicitly called by the caller; and we'd more likely be looking at something like `get_regex_for(field)`.